app/logic/courseManagement.py

```diff
@@ -49,33 +50,36 @@ def updateCourse(courseData):
         of instuctors assigned to the course and update the information in the db.
     """
-    course= Course.get_by_id(courseData['courseID'])
-    for toggler in ["regularOccurenceToggle", "slSectionsToggle", "permanentDesignation"]:
-        courseData.setdefault(toggler, "off")
+    try:
+        course= Course.get_by_id(courseData['courseID'])
+        for toggler in ["regularOccurenceToggle", "slSectionsToggle", "permanentDesignation"]:
+            courseData.setdefault(toggler, "off")
 
-    Course.update(
-        courseName=courseData["courseName"],
-        courseAbbreviation=courseData["courseAbbreviation"],
-        courseCredit=courseData["credit"],
-        isRegularlyOccuring=("on" in courseData["regularOccurenceToggle"]),
-        term=courseData['term'],
-        status=CourseStatus.SUBMITTED,
-        isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
-        serviceLearningDesignatedSections=courseData["slDesignation"],
-        isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
-    ).where(Course.id == course.id).execute()
+        Course.update(
+            courseName=courseData["courseName"],
+            courseAbbreviation=courseData["courseAbbreviation"],
+            courseCredit=courseData["credit"],
+            isRegularlyOccuring=("on" in courseData["regularOccurenceToggle"]),
+            term=courseData['term'],
+            status=CourseStatus.SUBMITTED,
+            isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
+            serviceLearningDesignatedSections=courseData["slDesignation"],
+            isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
+        ).where(Course.id == course.id).execute()
+        for i in range(1, 7):
+            (CourseQuestion.update(questionContent=courseData[f"{i}"])
+                        .where((CourseQuestion.questionNumber == i) &
+                               (CourseQuestion.course==course)).execute())
 
-    for i in range(1, 7):
-        (CourseQuestion.update(questionContent=courseData[f"{i}"])
-                    .where((CourseQuestion.questionNumber == i) &
-                           (CourseQuestion.course==course)).execute())
-
-    instructorList = []
-    if 'instructor[]' in courseData:
-        instructorList = courseData.getlist('instructor[]')
+        instructorList = []
+        if 'instructor[]' in courseData:
+            instructorList = courseData.getlist('instructor[]')
 
-    CourseInstructor.delete().where(CourseInstructor.course == course).execute()
-    for instructor in instructorList:
-        CourseInstructor.create(course=course, user=instructor)
+        CourseInstructor.delete().where(CourseInstructor.course == course).execute()
+        for instructor in instructorList:
+            CourseInstructor.create(course=course, user=instructor)
+        createLog(f"Edited SLC proposal: {courseData['courseName']}")
+    except:
+        return False;
```

Comment:
Found one last thing (probably). The development version of this function got rid of the blanket try/except altogether. If you are going to add it back in, you need to print the exception so we know what went wrong.